src/ophys_etl/modules/segmentation/filter/roi_filter.py

```diff
@@ -0,0 +1,253 @@
+from typing import List, Dict, Optional, Union
+from abc import ABC, abstractmethod
+import argschema
+import json
+import h5py
+import pathlib
+import numpy as np
+
+from ophys_etl.modules.decrosstalk.ophys_plane import OphysROI
+
+from ophys_etl.modules.segmentation.merge.roi_utils import (
+    ophys_roi_to_extract_roi)
+
+from ophys_etl.modules.segmentation.qc_utils.roi_comparison_utils import (
+    roi_list_from_file)
+
+
+class ROIBaseFilter(ABC):
+    """
+    Abstract base class for ROI filtering mechanisms.
+
+    Sub-classes of this calss need to implement
+    self.is_roi_valid which accepts an OphysROI and returns
+    a boolean indicating whether or not that ROI is accepted
+    as valid according to the filter's criterion.
+
+    The __init__ for the sub-class also needs to define
+    self._reason, a string summarizing what the filter does
+    so that users can know why their ROIs were flagged as
+    invalid.
+    """
+
+    _reason = None  # reason ROI is marked as invalid
```

Comment:
I think this should be defined in `__init__`. Then subclasses will have to pass it in using `super().__init__`

Reply:
I was trying to avoid requiring future developers from having to call `super()` in their sub-classes. It seems unnecessarily complicated to me in this case.

That being said: if we just did nothing with `_reason` in the base class (no default and no `__init__` in the base calss), developers who forget to set `_reason` in their sub-classes will get an attribute error when the `FilterRunner` tries to call `this_filter.reason`.

Does that seem like an acceptable solution?